### ISSUE TITLE
Bad instruction following in simulated Shanghai dialect conversations

### ISSUE THREAD

**Issue by LiuHeyang0114** [QwenLM/Qwen3-Omni#159]
### Description

Qwen3-Omni-30B-A3B-Instruct struggles with speech interactions in Shanghainese dialect scenarios. I tested it on two test sets: one with TeleVAL's Chinese QA in Shanghainese, and the other with Qwen-TTS synthesized Shanghainese dialect questions. Qwen3-Omni can generate speech responses in the former, although the accuracy is significantly lower than in Mandarin (which is unavoidable). However, in the Qwen-TTS synthesized dialect, it performs an ASR task and fails to generate an appropriate response. I have consulted with others who are familiar with Shanghainese and confirmed that the questions generated by Qwen-TTS are recognizable and of high quality.

### Reproduction

```
import soundfile as sf
from transformers import Qwen3OmniMoeForConditionalGeneration, Qwen3OmniMoeProcessor
from transformers.models.esm.openfold_utils import data_transforms
from qwen_omni_utils import process_mm_info
import json, os
import pdb
import argparse

def local_infer_wavdir(wav_dir, json_dir, output_wav_dir, output_json):

    start_id = 0
    if os.path.exists(output_json):
      with open(output_json, 'r', encoding = 'utf-8') as finished:
        data_finished = [json.loads(line) for line in finished]
        start_id = len(data_finished)

    MODEL_PATH = "../Qwen3-Omni-30B-A3B-Instruct"
    # MODEL_PATH = "Qwen/Qwen3-Omni-30B-A3B-Thinking"

    model = Qwen3OmniMoeForConditionalGeneration.from_pretrained(
        MODEL_PATH,
        dtype="auto",
        device_map="auto",
        attn_implementation="flash_attention_2",
    )

    processor = Qwen3OmniMoeProcessor.from_pretrained(MODEL_PATH)
    
    with open(json_dir, 'r', encoding='utf-8') as f:
        questions = json.load(f)


    for question in questions[start_id:]:
        audio_input = os.path.join(wav_dir, question['Audio'])
        save_path = os.path.join(output_wav_dir, question['Qid']+'.wav')
        if question.get('Context', None):
          conversation = []
          for turn in question['Context']:
            if turn['from'] == 'user':
              conversation.append({
                  "role": "user",
                  "content": turn["value"],
              })
            elif turn['from'] == 'assistant':
              conversation.append({
                  "role": "assistant",
                  "content": turn["value"],
              })
            else:
              raise ValueError
          conversation.append({
                  "role": "user",
                  "content": [
                      {"type": "audio", "audio": audio_input},
                  ],
              })
        else:
          conversation = [
              {
                  "role": "user",
                  "content": [
                      {"type": "audio", "audio": audio_input},
                    #   {"type": "text", "text": "你是一个语音对话助手，请用普通话回答这个问题。"}
                  ],
              },
          ]

        # Set whether to use audio in video
        USE_AUDIO_IN_VIDEO = True

        # Preparation for inference
        text = processor.apply_chat_template(conversation, add_generation_prompt=True, tokenize=False)
        audios, images, videos = process_mm_info(conversation, use_audio_in_video=USE_AUDIO_IN_VIDEO)
        inputs = processor(text=text, 
                          audio=audios, 
                          images=images, 
                          videos=videos, 
                          return_tensors="pt", 
                          padding=True, 
                          use_audio_in_video=USE_AUDIO_IN_VIDEO)
        inputs = inputs.to(model.device).to(model.dtype)

        # Inference: Generation of the output text and audio
        text_ids, audio = model.generate(**inputs, 
                                        speaker="Ethan", 
                                        thinker_return_dict_in_generate=True,
                                        use_audio_in_video=USE_AUDIO_IN_VIDEO)

        text = processor.batch_decode(text_ids.sequences[:, inputs["input_ids"].shape[1] :],
                                      skip_special_tokens=True,
                                      clean_up_tokenization_spaces=False)
        print(text)
        if audio is not None:
            sf.write(
                save_path,
                audio.reshape(-1).detach().cpu().numpy(),
                samplerate=24000,
            )
        
        ans_file = open(output_json, "a")
        # ans_file.write(json.dumps({"question_id": question['Qid'] , "prediction": prediction}) + "\n")
        ans_file.write(
            json.dumps(
                    {"question_id": question['Qid'], "prediction": text[0]},
                    ensure_ascii=False  # 关键：保留中文
            ) + "\n"
        )
        ans_file.close()


if __name__ == "__main__":

    parser = argparse.ArgumentParser(description='Qwen3-Omni Inference')
    parser.add_argument('--wav_dir', required=True, type=str)
    parser.add_argument('--json_dir', required=True, type=str)   
    parser.add_argument('--output_wav_dir', required=True, type=str)
    parser.add_argument('--output_json', required=True, type=str)   
    # parser.add_argument('--start_id', default=0, type=int) 
    args = parser.parse_args()
    os.makedirs(args.output_wav_dir, exist_ok=True)
    local_infer_wavdir(args.wav_dir, args.json_dir, args.output_wav_dir, args.output_json)


    # {
    #     "Qid": "KnowledgeQA_1",
    #     "Audio": "KnowledgeQA/KnowledgeQA_1.wav",
    #     "Question": "姆妈讲今朝小菜买得多个，夜饭吃勿脱个物事要摆到啥地方去？",
    #     "Question-zh": "妈妈说今天菜买多了，晚饭吃不完的东西要放到哪里去？",
    #     "Answer": "冰箱里"
    # },

    # {"question_id": "KnowledgeQA_1", "prediction": "姆妈讲今朝小菜买得多啊，夜饭吃不掉的么子要摆到啥地方去？"}

```

### Logs

```shell

```

### Environment Information

```
Package                           Version                           Editable project location
--------------------------------- --------------------------------- ------------------------------------------
accelerate                        1.10.1
aiohappyeyeballs                  2.6.1
aiohttp                           3.12.15
aiosignal                         1.4.0
airportsdata                      20250909
annotated-types                   0.7.0
anyio                             4.11.0
astor                             0.8.1
async-timeout                     5.0.1
attrs                             25.3.0
audioread                         3.0.1
av                                15.1.0
blake3                            1.0.6
cachetools                        6.2.0
certifi                           2025.8.3
cffi                              2.0.0
charset-normalizer                3.4.3
click                             8.3.0
cloudpickle                       3.1.1
cmake                             4.1.0
compressed-tensors                0.10.2
cupy-cuda12x                      13.6.0
decorator                         5.2.1
depyf                             0.18.0
dill                              0.4.0
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
einops                            0.8.1
email-validator                   2.3.0
exceptiongroup                    1.3.0
fastapi                           0.117.1
fastapi-cli                       0.0.13
fastapi-cloud-cli                 0.2.1
fastrlock                         0.8.3
filelock                          3.19.1
flash_attn                        2.8.3
frozenlist                        1.7.0
fsspec                            2025.9.0
gguf                              0.17.1
h11                               0.16.0
hf-xet                            1.1.10
httpcore                          1.0.9
httptools                         0.6.4
httpx                             0.28.1
huggingface-hub                   1.0.0rc1
idna                              3.10
interegular                       0.3.3
Jinja2                            3.1.6
jiter                             0.11.0
joblib                            1.5.2
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
lark                              1.2.2
lazy_loader                       0.4
librosa                           0.11.0
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.10.12
markdown-it-py                    4.0.0
MarkupSafe                        3.0.2
mdurl                             0.1.2
mistral_common                    1.8.5
mpmath                            1.3.0
msgpack                           1.1.1
msgspec                           0.19.0
multidict                         6.6.4
nest-asyncio                      1.6.0
networkx                          3.4.2
ninja                             1.13.0
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.6.4.1
nvidia-cuda-cupti-cu12            12.6.80
nvidia-cuda-nvrtc-cu12            12.6.77
nvidia-cuda-runtime-cu12          12.6.77
nvidia-cudnn-cu12                 9.5.1.17
nvidia-cufft-cu12                 11.3.0.4
nvidia-cufile-cu12                1.11.1.6
nvidia-curand-cu12                10.3.7.77
nvidia-cusolver-cu12              11.7.1.2
nvidia-cusparse-cu12              12.5.4.2
nvidia-cusparselt-cu12            0.6.3
nvidia-nccl-cu12                  2.26.2
nvidia-nvjitlink-cu12             12.6.85
nvidia-nvtx-cu12                  12.6.77
openai                            1.90.0
opencv-python-headless            4.12.0.88
outlines                          0.1.11
outlines_core                     0.1.26
packaging                         25.0
partial-json-parser               0.2.1.1.post6
pillow                            11.3.0
pip                               25.2
platformdirs                      4.4.0
pooch                             1.8.2
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.3.2
protobuf                          6.32.1
psutil                            7.1.0
py-cpuinfo                        9.0.0
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.23
pydantic                          2.11.9
pydantic_core                     2.33.2
pydantic-extra-types              2.10.5
Pygments                          2.19.2
python-dotenv                     1.1.1
python-json-logger                3.3.0
python-multipart                  0.0.20
PyYAML                            6.0.3
pyzmq                             27.1.0
qwen-omni-utils                   0.0.8
ray                               2.49.2
referencing                       0.36.2
regex                             2025.9.18
requests                          2.32.5
rich                              14.1.0
rich-toolkit                      0.15.1
rignore                           0.6.4
rpds-py                           0.27.1
safetensors                       0.6.2
scikit-learn                      1.7.2
scipy                             1.15.3
sentencepiece                     0.2.1
sentry-sdk                        2.39.0
setuptools                        78.1.1
setuptools-scm                    9.2.0
shellingham                       1.5.4
sniffio                           1.3.1
soundfile                         0.13.1
soxr                              1.0.0
starlette                         0.48.0
sympy                             1.14.0
threadpoolctl                     3.6.0
tiktoken                          0.11.0
tokenizers                        0.22.1
tomli                             2.2.1
torch                             2.7.0
torchaudio                        2.7.0
torchvision                       0.22.0
tqdm                              4.67.1
transformers                      4.57.0.dev0
triton                            3.3.0
typer                             0.19.2
typer-slim                        0.19.2
typing_extensions                 4.15.0
typing-inspection                 0.4.1
urllib3                           2.5.0
uvicorn                           0.37.0
uvloop                            0.21.0
vllm                              0.9.3.dev4+g6d8d50a01.precompiled MY_PATH/projects/Qwen3-Omni/vllm
watchfiles                        1.1.0
websockets                        15.0.1
wheel                             0.45.1
xformers                          0.0.30
xgrammar                          0.1.19
yarl                              1.20.1

```

### Known Issue

- [x] The issue hasn't been already addressed in Documentation, Issues, and Discussions.
